components/kyma-environment-broker/internal/process/upgrade_kyma/btp_operator_overrides.go

```diff
@@ -27,7 +29,8 @@ func (s *BTPOperatorOverridesStep) Name() string {
 }
 
 func (s *BTPOperatorOverridesStep) Run(operation internal.UpgradeKymaOperation, log logrus.FieldLogger) (internal.UpgradeKymaOperation, time.Duration, error) {
```

Comment:
check if clusterId is not empty,
If it is empty, use kubeconfig to fetch it, then
operation.clusterId = fetchedClusterID
operationManager.save(operation)